tests/components/datadog/test_init.py

```diff
@@ -145,3 +153,17 @@ async def test_state_changed(hass: HomeAssistant) -> None:
             hass.states.async_set("domain.test", invalid, {})
             await hass.async_block_till_done()
             assert not mock_statsd.gauge.called
+
+
+async def test_unload_entry(hass: HomeAssistant) -> None:
```

Comment:
This test should load the integration instead with mocking the lib and assert it from there.
We don't want to use the internals directly in tests.